packages/block-editor/src/store/selectors.js

```diff
@@ -3085,9 +3085,7 @@ export const getBlockEditingMode = createRegistrySelector(
 				const isContent = hasContentRoleAttribute( name );
 				return isContent ? 'contentOnly' : 'disabled';
 			}
-			// Otherwise, check if there's an ancestor that is contentOnly
-			const parentMode = getBlockEditingMode( state, rootClientId );
-			return parentMode === 'contentOnly' ? 'default' : parentMode;
+			return 'default';
```

Comment:
I personally found this code really confusingly written (and the comment made me more confused). It'd be easier to understand if it were written like this:
```js
const parentMode = getBlockEditingMode( state, rootClientId );
// Child blocks inherit a 'disabled' state from their parents.
if ( parentMode === 'disabled' ) {
    return 'disabled';
}

// Other modes are not inherited, return the 'default' value.
return 'default';
```

That would make it clear that the part I've refactored is the `if ( parentMode === 'disabled' )` block. It has now moved to the `withDerivedBlockEditingModes` reducer.

Reply:
I'm having a hard time connecting your comment with the existing code and why the change? (Agreed it's confusing)

Reply:
The handling of the inherited disabled states has been moved to `withDerivedBlockEditingModes`, this is the new code that replaces the recursive call in the selector:
https://github.com/WordPress/gutenberg/blob/0cef3d4ed56c279b5c6f9be0dbb46d531a137993/packages/block-editor/src/store/reducer.js#L2300-L2328

As for the why - it's related to what I mentioned comment about letting the explicit block editing modes override the derived ones. It would've been difficult to do that for the inherited disabled states because the code for determining that was in the selector.